### PR TITLE
add tps benchmark

### PR DESCRIPTION
**Pull request by jaffee**

### CODE REVIEW

**Comment by alanbernstein** [comment]
I tried this with an empty pilosa, and had a few problems - the benchmark handles the "no indexes" case, but it panics if it tries to run with one empty index. Then, running with one index with one empty field, it returns results that might not be super meaningful (maybe an upper bound on TPS?)

This isn't really a problem if you know what you're doing with the benchmark - maybe the description could be a little more explicit about the need to specify appropriate index and fields?

File: bench/bench.go [hunk 1]
```diff
@@ -294,3 +294,77 @@ func (s *Stats) Add(td time.Duration) {
...
+
+// NewNumStats gets a NumStats object.
+func NewNumStats() *NumStats {
+	return &NumStats{
+		Min: 1<<63 - 1,
+		All: make([]int64, 0),
+	}
+}
+
+// Add adds a new time to the stats object.
```
Comment (alanbernstein) [inline bench/bench.go]: time -> value (?)

File: bench/bench.go [hunk 2]
```diff
@@ -294,3 +294,77 @@ func (s *Stats) Add(td time.Duration) {
...
+}
+
+func (s *NumStats) Combine(other *NumStats) {
+	if other.Min < s.Min {
+		s.Min = other.Min
+	}
+	if other.Max > s.Max {
+		s.Max = other.Max
+	}
+	s.NumZero += other.NumZero
```
Comment (alanbernstein) [inline bench/bench.go]: What's this used for?

**Comment by jaffee** [comment]
good feedback, thanks. updated stuff.

File: bench/tps.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+			return b.runQueries(client, index, fields, queries, i, stats[i])
+		})
+	}
+	err = eg.Wait()
+	if err == nil {
+		for i := 1; i < len(stats); i++ {
+			stats[0].Combine(stats[i])
+		}
+		result.Extra["countstats"] = stats[0]
+		duration := time.Since(start)
```
Comment (alanbernstein) [inline bench/tps.go]: Can we move this to just after the `eg.Wait()`?

Comment (jaffee) [reply bench/tps.go]: oh yes, very much

**Pull request merged**